src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3004,6 +3005,7 @@ impl Coordinator {
             },
             TargetCluster::Active,
             None,
+            None,
```

Comment:
Why are we not enforcing limits here? Afaict this is the code path for `INSERT/UPDATE ... SELECT` and nothing prevents the SELECT from being expensive.

Reply:
+1

What's tricky here is the already existing variable `max_query_result_size != max_result_size`. The former limits how large of a result we'll return to the user, while the later limits how large of a result we'll allow `clusterd` to send to `environmentd`.

Some more context is [here](https://github.com/MaterializeInc/database-issues/issues/8747) but because the results of this peek are not generally user facing (except in the case where the user specifies a `RETURNING` clause) we don't want to set a `max_query_result_size`.